controlplane/kubeadm/internal/cluster.go

```diff
@@ -135,30 +136,27 @@ func (m *ManagementCluster) GetEtcdCerts(ctx context.Context, cluster types.Name
 	if err := m.Client.Get(ctx, etcdCAObjectKey, etcdCASecret); err != nil {
 		return nil, nil, errors.Wrapf(err, "failed to get secret; etcd CA bundle %s/%s", etcdCAObjectKey.Namespace, etcdCAObjectKey.Name)
 	}
-	etcdCACertData, ok := etcdCASecret.Data[secret.TLSCrtDataName]
+	crtData, ok := etcdCASecret.Data[secret.TLSCrtDataName]
 	if !ok {
-		return nil, nil, errors.New("empty ca certificate")
+		return nil, nil, errors.Errorf("etcd tls crt does not exist for cluster %s/%s", cluster.Namespace, cluster.Name)
```

Comment:
+1 to the `namespace/name` convention for errors.

All errors should have this information--no one wants to guess what the resource was.  I see we use it throughout CAPI. I wonder if we want a linter rule?

Reply:
we probably don't need a linter yet unless it's a sufficiently smart rule as some errors just won't have a namespace/name to report

Reply:
related, i added this information to all of the errors where it made sense

Reply:
I'm wondering if we can leverage the structured logging for this instead of having to embed namespace/name in the error messages again.

Reply:
That would be nicer... maybe let's revisit as we get a chance to test it all out and see how the errors feel vs logging? I'm not opposed to either way though.